src/helm/proxy/clients/chat_gpt_client.py

```diff
@@ -0,0 +1,98 @@
+    We use the unofficial ChatGPT Python client: https://github.com/acheong08/ChatGPT.
+    """
+
+    REQUEST_TIMEOUT_SECONDS: int = 10 * 60  # 10 minutes
+
+    def __init__(
+        self, email: str, password: str, lock_file_path: str, cache_config: CacheConfig, tokenizer_client: Client
+    ):
+        self.chat_bot = Chatbot({"email": email, "password": password}, debug=True)
+        self.tokenizer_client: Client = tokenizer_client
+        self.cache = Cache(cache_config)
+
+        # Since we want a brand new chat session per request, only allow a single request at a time.
+        self._lock = FileLock(lock_file_path, timeout=ChatGPTClient.REQUEST_TIMEOUT_SECONDS)
+
+    def make_request(self, request: Request) -> RequestResult:
+        completions: List[Sequence] = []
+        all_cached = True
+        request_time = 0
+        request_datetime: Optional[int] = None
+
+        # `num_completions` is not supported, so instead make `num_completions` separate requests.
+        for completion_index in range(request.num_completions):
+            try:
+
+                def do_it():
+                    with self._lock:
+                        self.chat_bot.refresh_session()
+                        result: Dict[str, Any] = self.chat_bot.get_chat_response(request.prompt, output="text")
+                        assert "message" in result, f"Invalid response: {result}"
+                        self.chat_bot.reset_chat()
+                        return result
+
+                raw_request: Dict[str, Any] = {
+                    "model": request.model_engine,
+                    "prompt": request.prompt,
+                    "completion_index": completion_index,
+                }
+                cache_key = Client.make_cache_key(raw_request, request)
+                response, cached = self.cache.get(cache_key, wrap_request_time(do_it))
+            except Exception as e:
+                error: str = f"ChatGPT error: {e}"
+                return RequestResult(success=False, cached=False, error=error, completions=[], embedding=[])
+
+            # The ChatGPT API doesn't support echo. If `echo_prompt` is true, combine the prompt and completion.
+            text: str = request.prompt + response["message"] if request.echo_prompt else response["message"]
+            # The ChatGPT API doesn't return us tokens or logprobs, so we tokenize ourselves.
+            tokenization_result: TokenizationRequestResult = self.tokenize(
+                # We're assuming ChatGPT uses the GPT-2 tokenizer.
+                TokenizationRequest(text, tokenizer="huggingface/gpt2")
+            )
+            completion = Sequence(
+                text=response["message"],
+                # Log probs are not currently not supported by the ChatGPT, so set to 0 for now.
+                logprob=0,
+                tokens=[
+                    Token(text=str(text).replace("Ġ", " "), logprob=0, top_logprobs={})
```

Comment:
Can we factor the replace logic out?

(@yifanmai : this is an example where top_logprobs isn't set, so maybe we should just remove the warning to reduce noise?)